aspnet/2-structured-data/Models/DbBookStore.cs

```diff
@@ -46,12 +46,13 @@ public void Delete(long id)
         // [START list]
         public BookList List(int pageSize, string nextPageToken)
         {
-            var pageOfBooks = (null == nextPageToken ?
-                (from book in _dbcontext.Books orderby book.Id select book) :
-                (from book in _dbcontext.Books
-                 where book.Id > Int32.Parse(nextPageToken)
-                 orderby book.Id
-                 select book)).Take(pageSize + 1);
+            IQueryable<Book> query = from book in _dbcontext.Books orderby book.Id select book;
+            if (nextPageToken != null)
+            {
+                long previousBookId = long.Parse(nextPageToken);
+                query = from book in query where book.Id > previousBookId select book;
+            }
+            var pageOfBooks = query.Take(pageSize + 1);

```

Comment:
For code sample simplicity, we've been foregoing this extra piece of logic in Bookshelf samples, eg. [node](https://github.com/GoogleCloudPlatform/nodejs-getting-started/blob/master/2-structured-data/books/model-cloudsql.js#L37), [python](https://github.com/GoogleCloudPlatform/getting-started-python/blob/master/2-structured-data/bookshelf/model_cloudsql.py#L62), [ruby](https://github.com/GoogleCloudPlatform/getting-started-ruby/blob/master/2-cloud-sql/app/controllers/books_controller.rb#L21-L22)

(*true - this means that if 10 books are returned, a 'more' link will be displayed even if there is not an 11th book.  but we've favored the simpler logic for the code sample so that the `+1 "peeking"` does not need to be explained.*)